clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/view/OpenstackSecurityGroupProvider.groovy

```diff
@@ -17,55 +17,34 @@
 
 package com.netflix.spinnaker.clouddriver.openstack.provider.view
 
-import com.netflix.spinnaker.clouddriver.model.AddressableRange
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.cats.cache.Cache
+import com.netflix.spinnaker.cats.cache.CacheData
+import com.netflix.spinnaker.cats.cache.RelationshipCacheFilter
 import com.netflix.spinnaker.clouddriver.model.SecurityGroupProvider
-import com.netflix.spinnaker.clouddriver.model.securitygroups.IpRangeRule
-import com.netflix.spinnaker.clouddriver.model.securitygroups.Rule
 import com.netflix.spinnaker.clouddriver.openstack.OpenstackCloudProvider
-import com.netflix.spinnaker.clouddriver.openstack.client.OpenstackClientProvider
+import com.netflix.spinnaker.clouddriver.openstack.cache.Keys
 import com.netflix.spinnaker.clouddriver.openstack.model.OpenstackSecurityGroup
-import com.netflix.spinnaker.clouddriver.openstack.security.OpenstackNamedAccountCredentials
-import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider
 import groovy.util.logging.Slf4j
-import org.openstack4j.model.compute.SecGroupExtension
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Component
 
+import static com.netflix.spinnaker.clouddriver.openstack.cache.Keys.Namespace.SECURITY_GROUPS
+
 /**
  * Provides a view of existing Openstack security groups in all configured Openstack accounts.
- *
- * TODO: Remove direct lookup to Openstack and lookup from cache view instead
  */
 @Slf4j
 @Component
 class OpenstackSecurityGroupProvider implements SecurityGroupProvider<OpenstackSecurityGroup> {
 
-  final AccountCredentialsProvider accountCredentialsProvider
+  final Cache cacheView
+  final ObjectMapper objectMapper
 
   @Autowired
-  OpenstackSecurityGroupProvider(AccountCredentialsProvider accountCredentialsProvider) {
-    this.accountCredentialsProvider = accountCredentialsProvider
-  }
-
-  Set<OpenstackSecurityGroup> getSecurityGroups(boolean includeRules) {
-    // TODO This is not thread safe and not cached
-    // This just proves the provider is working once it has a data set.
-    Set<OpenstackSecurityGroup> securityGroups = []
-    accountCredentialsProvider.all.each { account ->
-      if (account instanceof OpenstackNamedAccountCredentials) {
-        OpenstackClientProvider provider = ((OpenstackNamedAccountCredentials) account).credentials.provider
-
-        provider.allRegions.each { region ->
-          securityGroups.addAll(
-            provider.getSecurityGroups(region)
-              .flatten()
-              .collect { convertToOpenstackSecurityGroup(includeRules, it, account.name, region) }
-          )
-        }
-      }
-    }
-
-    securityGroups
+  OpenstackSecurityGroupProvider(Cache cacheView, ObjectMapper objectMapper) {
```

Comment:
Just ensures that constructor args can't be changed, so in our case nothing.  It would be more of a conventional thing and since we aren't doing that all over the place, lets forget it.